localstripe/resources.py

```diff
@@ -3276,7 +3276,10 @@ def __init__(self, country=None, customer=None, type=None, value=None,
             assert type in ('eu_vat', 'nz_gst', 'au_abn')
             assert _type(value) is str and len(value) > 10
             if country is None:
-                country = value[0:2]
+                if type == 'eu_vat':
+                    country = value[0:2]
+                else:
+                    country = type[0:2].upper()
```

Comment:
Suggestion:
```suggestion
                if type == 'eu_vat':
                    country = value[0:2]
                elif type in ('nz_gst', 'au_abn'):
                    country = type[0:2].upper()
                else:
                    raise
```

The goal is to avoid "missing" cases if one day we add another value in the list `('eu_vat', 'nz_gst', 'au_abn')`.

(and `nz_gst, au_abn` → `nz_gst, au_abn, es_cif` in the next commit)

Reply:
Ok interesting ! Do you really suggest adding a naked return for "missing" case (I thought this was only for re-raising an exception in an `except` block), or do you mean something like `raise UserError(500, 'Not implemented')` (that we're already using elsewhere in the codebase) ?

Reply:
You're right, `raise` is a bit brutal, and there is no exception to reraise...

`raise UserError` wouldn't be suited, because it's not a user error, it would be a bug on our side (because we already checked `type` is amongst valid values a few lines above). We would need something like `raise ABugWasIntroducedInOurCodeAndNeedsToBeFixed()`. Since this doesn't exist (yet) in localstripe, I propose `assert False  # shouldn't happen because type is checked above`. What do you think?

Reply:
>  I propose assert False  # shouldn't happen because type is checked above. 

Ok, done in last push.